toolchain/check/import_cpp.cpp

```diff
@@ -22,28 +23,26 @@ namespace Carbon::Check {
 
 // Generates C++ file contents to #include all requested imports.
 static auto GenerateCppIncludesHeaderCode(
-    llvm::ArrayRef<std::pair<llvm::StringRef, SemIRLoc>> imports)
+    Context& context, llvm::ArrayRef<Parse::Tree::PackagingNames> imports)
     -> std::string {
   std::string code;
   llvm::raw_string_ostream code_stream(code);
-  for (const auto& [path, _] : imports) {
-    code_stream << "#include \"" << FormatEscaped(path) << "\"\n";
+  for (const Parse::Tree::PackagingNames& import : imports) {
+    code_stream << "#include \""
+                << FormatEscaped(
+                       context.string_literal_values().Get(import.library_id))
+                << "\"\n";
   }
   return code;
 }
 
-auto ImportCppFiles(
-    Context& context, llvm::StringRef importing_file_path,
-    llvm::ArrayRef<std::pair<llvm::StringRef, SemIRLoc>> imports,
-    llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs) -> void {
-  size_t num_imports = imports.size();
-  if (num_imports == 0) {
-    return;
-  }
-
+static auto GenerateAst(Context& context, llvm::StringRef importing_file_path,
+                        llvm::ArrayRef<Parse::Tree::PackagingNames> imports,
+                        llvm::IntrusiveRefCntPtr<llvm::vfs::FileSystem> fs)
+    -> std::optional<std::unique_ptr<clang::ASTUnit>> {
```

Comment:
Given this is a pointer, had you considered using nullptr instead of  `std::optional`? I think that's what we're doing elsewhere, because it reduces the amount of type wrapping. We generally try not to assume that pointers are non-null, and only use `optional` when there's not another way to indicate "none".

```suggestion
    -> std::unique_ptr<clang::ASTUnit> {
```

But more than that, is it worth trying to always return an AST? i.e., even if the C++ code has an error, do you think it's worth trying to emit diagnostics where Carbon code touches C++ code which successfully produced AST nodes, as part of error recovery during diagnosis?

Reply:
I've changed that to always return an AST if Clang managed to generate one, even if there are errors.
If we find this is cause more harm than good we can change that appraoch.